ocaml/typing/jkind.ml

```diff
@@ -1005,13 +1165,63 @@ module Jkind_desc = struct
     let bits64 = of_const Const.Primitive.bits64.jkind
   end
 
+  let product jkinds =
```

Comment:
This is very similar to `jkind_of_product_annotations`. Combine?

Reply:
Sadly they look more similar than they are - they are working on different types with similar names.  I did not find a clean way to combine them.